### ISSUE TITLE
Train custom dataset ！！！

### ISSUE THREAD

**Issue by Double-zh** [ifzhang/ByteTrack#41]
“”“”“”“”“”“”“”“”“”“”
First, you need to prepare your dataset in COCO format. You can refer to MOT-to-COCO or CrowdHuman-to-COCO. Then, you need to create a Exp file for your dataset. You can refer to the CrowdHuman training Exp file. Don't forget to modify get_data_loader() and get_eval_loader in your Exp file. Finally, you can train bytetrack on your dataset by running:
python3 tools/train.py -f exps/example/mot/your_exp_file.py -d 8 -b 48 --fp16 -o -c pretrained/yolox_x.pth
“”“”“”“”“”“”“”“”“”“”
Have you modified the yolox source code? Can you provide a modified file that can be trained directly (training exp file, get_data_loader () and get_eval_loader)

**Comment by ImSuMyatNoe** [comment]
@Double-zh May I know how to train the custom dataset? As my datasets are COCO format and json annotation format. I do not have frame id and track id like MOT17 (the example in this repository). How do I create the custom dataset format? Could you please tell me how to do it as I am newbie in this field:(

**Comment by Double-zh** [comment]
I haven't figured it out yet, waiting for your good news

> @Double-zh May I know how to train the custom dataset? As my datasets are COCO format and json annotation format. I do not have frame id and track id like MOT17 (the example in this repository). How do I create the custom dataset format? Could you please tell me how to do it as I am newbie in this field:(

**Comment by dddmmmyyy1998** [comment]
我认为他只需要MOT17 Train文件中的数据集格式。不需要知道每一帧中Id的值，他并没有对跟踪数据进行训练，只是对检测数据进行了训练，他的跟踪应该是无监督的跟踪

**Comment by wijjj** [comment]
> 

I think so, too. It's just the massive amount of data + input & val training image size that does the trick. I'm currently evaluating this approach: https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch

**Comment by wijjj** [comment]
> “”“”“”“”“”“”“”“”“”“” First, you need to prepare your dataset in COCO format. You can refer to MOT-to-COCO or CrowdHuman-to-COCO. Then, you need to create a Exp file for your dataset. You can refer to the CrowdHuman training Exp file. Don't forget to modify get_data_loader() and get_eval_loader in your Exp file. Finally, you can train bytetrack on your dataset by running: python3 tools/train.py -f exps/example/mot/your_exp_file.py -d 8 -b 48 --fp16 -o -c pretrained/yolox_x.pth “”“”“”“”“”“”“”“”“”“” Have you modified the yolox source code? Can you provide a modified file that can be trained directly (training exp file, get_data_loader () and get_eval_loader)

Yeah I needed to write another script in order to create the kind of dataset separation that BT requires. Still had some issues with (sub-)directories and wrong way of splitting data at first.
